src/test/java/org/stellar/sdk/Sep10ChallengeTest.java

```diff
@@ -703,12 +740,35 @@ public void testReadChallengeTransactionInvalidDataValueWrongEncodedLength() thr
       Sep10Challenge.readChallengeTransaction(challenge, server.getAccountId(), Network.TESTNET, domainName);
       fail();
     } catch (InvalidSep10ChallengeException e) {
-      assertEquals("Random nonce encoded as base64 should be 64 bytes long.", e.getMessage());
+      assertEquals("Random nonce before encoding as base64 should be 48 bytes long.", e.getMessage());
     }
   }
 
   @Test
-  public void testReadChallengeTransactionInvalidDataValueCorruptBase64() throws IOException {
+  public void testReadChallengeTransactionValidDoesNotVerifyHomeDomain() throws IOException, InvalidSep10ChallengeException {
+    KeyPair server = KeyPair.random();
+    KeyPair client = KeyPair.random();
+    Network network = Network.TESTNET;
+    String domainName = "example.com";
+
+    long now = System.currentTimeMillis() / 1000L;
+    long end = now + 300;
+    TimeBounds timeBounds = new TimeBounds(now, end);
+
+    Transaction transaction = Sep10Challenge.newChallenge(
+        server,
+        network,
+        client.getAccountId(),
+        domainName,
+        timeBounds
+    );
+
+    Sep10Challenge.ChallengeTransaction challengeTransaction = Sep10Challenge.readChallengeTransaction(transaction.toEnvelopeXdrBase64(), server.getAccountId(), Network.TESTNET, null);
+    assertEquals(new Sep10Challenge.ChallengeTransaction(transaction, client.getAccountId()), challengeTransaction);
```

Comment:
Why do we need a `ChallengeTransaction` class? I'm not saying we should remove it but I was originally confused on the difference between `Transaction` and `ChallengeTransaction`, since that distinction doesn't exist in the other SDK's. 

It appears to be a collection of helper functions common to challenge transaction as opposed to normal transactions.

Reply:
Maybe the class name is not so appropriate that makes you confused.

We need it because Java does not support multiple return values.